negative array index should not compile

V version: **0.1.18**
OS: **Ubuntu 18.04.2 LTS (Bionic Beaver)** 

### What did you do?

```
fn fibo4m(nums mut []int, limit int) []int {
    if nums[-1] > limit {
        return nums
    }
    nums << nums[-2] + nums[-1]
    return fibo4m(mut nums, limit)
}
```

### What did you expect to see?

No error (or not a c error)

 
### What did you see instead?

>.ex2.c: In function ‘fibo4m’:
.ex2.c:3487:9: error: incompatible types when returning type ‘array_int * {aka struct array *}’ but ‘array_int {aka struct array}’ was expected
 return  nums ;
         ^...
(Use `v -debug` to print the entire error message)
>
> V panic: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -3352,7 +3352,11 @@ fn (p mut Parser) return_st() {
 				}
 
 				if total_text == '' || expr_type == 'void*' {
-					p.cgen.resetln('return $ret')
+					if expr_type == '${p.cur_fn.typ}*' {
+						p.cgen.resetln('return *$ret')
+					} else {
+						p.cgen.resetln('return $ret')
+					}
 				}  else {
 					tmp := p.get_tmp()
 					p.cgen.resetln('$expr_type $tmp = $ret;\n')

